dashboard/src/containers/App.js

```diff
@@ -55,13 +74,49 @@ class App extends React.Component {
         </div>
       </div>
     );
-  };
-}
+  },
+
+  componentDidMount() {
+    this.props.initialServerState();
+    let id = this.props.match.params.id;
+    if (id !== undefined) {
+      this.props.getQuery(id);
+    }
+  },
+
+  componentWillReceiveProps(nextProps) {
+    if (!nextProps.found) {
+      // Lets auto close the alert after 2 secs.
```

Comment:
So, if query is not found, we set the state as 'found' in 3 seconds. This seems confusing. Could we make it more sensical with a better naming maybe?